src/integration/test_suites/smoke_test_fc_ratchet_seed_write_read/smoke_test_fc_ratchet_seed_write_read.c

```diff
@@ -0,0 +1,404 @@
+//********************************************************************************
+// SPDX-License-Identifier: Apache-2.0
+//
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//********************************************************************************
+#include <string.h>
+#include <stdint.h>
+#include <time.h>
+#include <stdlib.h>
+
+#include "soc_address_map.h"
+#include "printf.h"
+#include "riscv_hw_if.h"
+#include "soc_ifc.h"
+#include "caliptra_ss_lc_ctrl_address_map.h"
+#include "caliptra_ss_lib.h"
+#include "fuse_ctrl.h"
+#include "lc_ctrl.h"
+#include "fuse_ctrl_mmap.h"
+
+volatile char* stdout = (char *)SOC_MCI_TOP_MCI_REG_DEBUG_OUT;
+#ifdef CPT_VERBOSITY
+    enum printf_verbosity verbosity_g = CPT_VERBOSITY;
+#else
+    enum printf_verbosity verbosity_g = LOW;
+#endif
+
+/**
+   This tests:
+   1 - writes to each ratchet seed one by one and reads back the value written was returned.
+   2 - calculates the digest for all HEKs
+   3 - tries writing and reading and checking a DAI error is returned
+   4 - zeroize all the HEK fuses and markers
+ */
+
+void ratchet_seed_write_read(void) {
+    uint32_t r_data[2];
+    uint32_t w_data[2];
+
+    w_data[0] = 0x3;
+    w_data[1] = 0x3;
+
+
+    //=================================
+    // Write - Read the ratchet seeds:
+    //=================================
+    VPRINTF(LOW, "==========================================\n");
+    VPRINTF(LOW, "Doing Write/Read of HEKs:\n");
+    VPRINTF(LOW, "==========================================\n");
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_0].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_0].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
```

Comment:
Doesn't this lead to an error? I'm surprised because as you `dai_wr()` with `granularity=32`, the second 32 bit word (`w_data[1]`) does not get written.